tests/integration/sdk_api/projects/test_project.py

```diff
@@ -12,16 +12,133 @@
+                )
+
+        # create several artifacts
+        artifact = {
+            "test": "artifact",
+            "labels": labels,
+            "status": {"bla": "blabla"},
+        }
+        artifact_keys = ["artifact_key_1", "artifact_key_2", "artifact_key_3"]
+        for artifact_key in artifact_keys:
+            db.store_artifact(
+                artifact_key,
+                artifact,
+                "some_uid",
+                tag="some_tag",
+                project=project.metadata.name,
+            )
+
+        projects = db.list_projects()
+        assert len(projects) == 1
+        assert projects[0].metadata.name == project_name
+
+        # verify artifacts and functions were created
+        project_artifacts = project.list_artifacts()
+        loaded_project_artifacts = projects[0].list_artifacts()
+        assert len(project_artifacts) == len(artifact_keys)
+        assert len(project_artifacts) == len(loaded_project_artifacts)
+        assert project_artifacts == loaded_project_artifacts
+
+        project_functions = project.list_functions()
+        loaded_project_functions = projects[0].list_functions()
+        assert len(project_functions) == len(function_names) * len(function_tags)
+        assert len(project_functions) == len(loaded_project_functions)
+
+        loaded_project_functions_dicts = [
+            func.to_dict() for func in loaded_project_functions
+        ]
+        assert all(
+            [
+                func.to_dict() in loaded_project_functions_dicts
+                for func in project_functions
+            ]
+        )
+
+        old_creation_time = projects[0].metadata.created
+
+        mlrun.new_project(project_name, overwrite=True)
+        projects = db.list_projects()
+        assert len(projects) == 1
+        assert projects[0].metadata.name == project_name
+
+        # ensure cascade deletion strategy
+        assert projects[0].list_artifacts() == []
+        assert projects[0].list_functions() is None
+        assert projects[0].metadata.created > old_creation_time
+
+    def test_overwrite_empty_project(self):
+        project_name = "some-project"
+
+        mlrun.new_project(project_name)
+        db = mlrun.get_run_db()
+
+        projects = db.list_projects()
+        assert len(projects) == 1
+        assert projects[0].metadata.name == project_name
+
+        assert projects[0].list_artifacts() == []
+        assert projects[0].list_functions() is None
+        old_creation_time = projects[0].metadata.created
+
+        # overwrite empty project
+        mlrun.new_project(project_name, overwrite=True)
+        projects = db.list_projects()
+        assert len(projects) == 1
+        assert projects[0].metadata.name == project_name
+
+        assert projects[0].list_artifacts() == []
+        assert projects[0].list_functions() is None
+        assert projects[0].metadata.created > old_creation_time
+
     def test_load_project_from_db(self):
```

Comment:
Per my other comment - let's also test that if you try to create a project from an invalid template (malformed file or non-existing file) you fail, but your old project remains.